CodeStyleAnalyzers/InterfaceAnalyzers/PublicModifier/PublicModifierInterfaceCodeFix.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace CodeStyleAnalyzers.InterfaceAnalyzers.PublicModifier
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public class PublicModifierInterfaceCodeFix : CodeFixProvider
+    {
+        private static readonly string CodeFixTitle = "Remove unnecessary access modifier";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(RuleConstants.DiagnosticId);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            var syntaxToken = root.FindToken(diagnosticSpan.Start);
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title: CodeFixTitle,
+                    createChangedDocument: cancellationToken => FixCode(context.Document, syntaxToken, cancellationToken),
+                    equivalenceKey: CodeFixTitle),
+                diagnostic);
+        }
+
+        private async Task<Document> FixCode(Document document, SyntaxToken syntax, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var nextToken = syntax.GetNextToken();
+            return document.WithSyntaxRoot(root.ReplaceTokens(new[] { syntax, nextToken }, (current, _)
```

Comment:
I'd perform the search/replace on the parent node of the token instead of the whole syntax root.

```cs
var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
var nextToken = syntax.GetNextToken();
editor.ReplaceNode(syntax.Parent, syntax.Parent.ReplaceTokens(new[] { syntax, nextToken }, (current, _) => ExtractSyntaxToken(syntax, current, nextToken)));
return editor.GetChangedDocument();
```

Reply:
[Perfom actions on parent node](https://github.com/messerli-informatik-ag/code-style/pull/63/commits/4ecf00e6191b05399e50e6ad0a0080d9985f921a)